core/src/test/java/org/infinispan/xsite/XSiteMergePolicyUnitTest.java

```diff
@@ -0,0 +1,216 @@
+package org.infinispan.xsite;
+
+import static org.testng.AssertJUnit.assertEquals;
+
+import java.util.Random;
+
+import org.infinispan.test.AbstractInfinispanTest;
+import org.infinispan.test.TestingUtil;
+import org.infinispan.util.concurrent.CompletionStages;
+import org.infinispan.xsite.spi.AlwaysRemoveXSiteEntryMergePolicy;
+import org.infinispan.xsite.spi.DefaultXSiteEntryMergePolicy;
+import org.infinispan.xsite.spi.PreferNonNullXSiteEntryMergePolicy;
+import org.infinispan.xsite.spi.PreferNullXSiteEntryMergePolicy;
+import org.infinispan.xsite.spi.SiteEntry;
+import org.infinispan.xsite.spi.XSiteEntryMergePolicy;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+/**
+ * Unit test for all provided {@link XSiteEntryMergePolicy}.
+ *
+ * @author Pedro Ruivo
+ * @since 12.0
+ */
+@Test(groups = "unit", testName = "xsite.XSiteMergePolicyUnitTest")
+public class XSiteMergePolicyUnitTest extends AbstractInfinispanTest {
```

Comment:
This test seems a bit over-engineered to me:

* The Sync class could have been just a method `merge(XSiteEntryMergePolicy, SiteEntry, SiteEntry)`
* The data providers could have been inlined, calling `doTest()` with the parameters explicitly instead of wrapping them in an `Object[][]` first.
   Each method has its own data provider, so reuse is not a factor.